public/enhanced-table-vis-controller.js

```diff
@@ -308,12 +308,30 @@ module.controller('EnhancedTableVisController', function ($scope, Private, confi
 
       let removedCounter = 0;
       _.forEach(hiddenColumns, function (item) {
-        let index = getRealColIndex(parseInt(item), splitColIndex);
-        table.columns.splice(index - removedCounter, 1);
-        _.forEach(table.rows, function (row) {
-          row.splice(index - removedCounter, 1);
-        });
-        removedCounter++;
+        try {
+          if (parseInt(item).toString() == item) {
+            let index = getRealColIndex(parseInt(item), splitColIndex);
+            let colToRemove = index - removedCounter;
+            console.log(colToRemove)
+            table.columns.splice(colToRemove, 1);
+            _.forEach(table.rows, function (row) {
+              row.splice(colToRemove, 1);
+            });
+            removedCounter++;
+          } else {
+            let itemIndex = findColIndexByTitle(table.columns, item, item, splitColIndex);
+            let index = getRealColIndex(itemIndex, splitColIndex);
+            let colToRemove = index;
+            console.log(colToRemove)
+            table.columns.splice(colToRemove, 1);
+            _.forEach(table.rows, function (row) {
+              row.splice(colToRemove, 1);
+            });
```

Comment:
These 4 lines are identical in the `if` and `else` sections.
Can you put this code after if/else statements to avoid duplicate code?

Reply:
I can but then I'll have to add a duplicate of `(parseInt(item).toString() == item)` to decide whether to increase `removedCounter` or not. Ideas?

Reply:
Actually I just found a little bug in using a mix of column indeces and column labels. 
In short, removedCounter++ should be kept in both cases but it can create some problems if you remove a column "test_column" (which has index X, say 2) and then you remove column with index Y<X, say 1. In that case, removedCounter would be increased to 1 at the time of removing "test_column". Thus, when it comes to remove column at index 1, it'll remove column at index colToRemove = 1 - 1 = 0, which is wrong. 
Though, it is a potential bug already present in the original release and it's due to the fact of not putting the column in order (if you write "4,5,2,3" in Hidden Columns, you'll see the behavior won't be the expected one). 
I'll see what I can do about it tomorrow. 
In case we do not find an elegant solution we can always put a disclaimer which specifies to use EITHER the column indeces OR the column_labels and, in case someone wants to use the indeces, to write them in ascending order.

Reply:
Ok this should be fixed.